providers/redfish/firmware.go

```diff
@@ -197,6 +197,71 @@ func (c *Conn) firmwareUpdateCompatible(ctx context.Context) (err error) {
 	return nil
 }
 
+// pipeReaderFakeSeeker wraps the io.PipeReader and implements the io.Seeker interface
+// to meet the API requirements for the Gofish client https://github.com/stmcginnis/gofish/blob/46b1b33645ed1802727dc4df28f5d3c3da722b15/client.go#L434
+//
+// The Gofish method linked does not currently perform seeks and so a PR will be suggested
+// to change the method signature to accept an io.Reader instead.
+type pipeReaderFakeSeeker struct {
+	*io.PipeReader
+}
+
+// Seek impelements the io.Seeker interface only to panic if called
+func (p pipeReaderFakeSeeker) Seek(offset int64, whence int) (int64, error) {
+	panic("Seek() not implemented for fake pipe reader seeker.")
+}
+
+// multipartPayloadSize prepares a temporary multipart form to determine the form size
+func multipartPayloadSize(payload []map[string]io.Reader) (int64, *bytes.Buffer, error) {
+	body := &bytes.Buffer{}
+	form := multipart.NewWriter(body)
+
+	var size int64
+	var err error
+	for idx, elem := range payload {
+		for key, reader := range elem {
+			var part io.Writer
+			if file, ok := reader.(*os.File); ok {
+				// Add update file fields
+				if _, err = form.CreateFormFile(key, filepath.Base(file.Name())); err != nil {
+					return 0, body, err
+				}
+
+				// determine file size
+				finfo, err := file.Stat()
+				if err != nil {
+					return 0, body, err
+				}
+
+				size = finfo.Size()
```

Comment:
gotcha, is that described somewhere in this repo (docs maybe)? Either way it would be nice to have a check in this loop maybe to make it obvious to code readers.

Reply:
I took a slightly different route and simplified the implementation - [059b754](https://github.com/bmc-toolbox/bmclib/pull/345/commits/059b754ae098c55eb3b2466e5aaf849e131ea974)

Instead of expecting a payload `[]map[string]io.Reader` its now the `multipartPayload` struct type,
the loop is also unrolled and its simpler to follow the form payload being built.